Slightly refactor PF endcap cluster calibration

guitargeek opened the pull request:
#### PR description:

In the PFEgammaAlgo, there is still this comment about code "stolen" from the PFECALSuperClusterAlgo:
https://github.com/cms-sw/cmssw/blob/master/RecoParticleFlow/PFProducer/src/PFEGammaAlgo.cc#L1774

I was thinking of ways to avoid this duplication, but it's not quite trivial because the code diverged a bit. However, a bit part of the duplicate code serves to compute cluster energy corrections, so I could move this into the PFEnergyCalibration class which is used by both algos. I think this is a good step to avoid this code duplication, but I don't want to go further for now to not risk accidentally changing the logic. What do you think?

#### PR validation:

CMSSW compiles and matrix tests pass.

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

No backport intended.

guitargeek committed:
don't store vectors just to accumulate them

guitargeek committed:
factored out calibrateEndcapClusterEnergies function

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-28960/13786

- This PR adds an extra 52KB to repository

cmsbuild commented:
A new Pull Request was created by  @guitargeek (Jonas Rembser) for master.

It involves the following packages:

RecoEcal/EgammaClusterAlgos
RecoParticleFlow/PFClusterTools
RecoParticleFlow/PFProducer
RecoParticleFlow/PFTracking


@perrotta, @cmsbuild, @slava77 can you please review it and eventually sign? Thanks.
@mmarionncern, @Sam-Harper, @makortel, @jainshilpi, @afiqaize, @rovere, @lgray, @sobhatta, @bachtis, @lecriste, @hatakeyamak, @argiro, @varuns23, @seemasharmafnal, @cbernet this is something you requested to watch as well.
@davidlange6, @silviodonato, @fabiocos you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 876e1dc08a9720e45fbefa6f681fbdc8101aafb5
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ab1c14/4654/summary.html
CMSSW: CMSSW_11_1_X_2020-02-13-1100
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ab1c14/4654/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 34
 - DQMHistoTests: Total histograms compared: 2694005
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2693658
 - DQMHistoTests: Total skipped: 346
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 33 files compared)
 - Checked 147 log files, 16 edm output root files, 34 DQM output files

silviodonato commented:
Do you have any comments? @perrotta @slava77

perrotta commented on RecoParticleFlow/PFClusterTools/src/PFEnergyCalibration.cc:
```diff
@@ -161,6 +161,59 @@ void PFEnergyCalibration::initializeCalibrationFunctions() {
+  double ps2_energy_sum = 0.;
+  double ePS1 = 0.;
+  double ePS2 = 0.;
+  int condP1 = 1;
```
Maybe if you make these two some "bool" variable, everything gets more intuitive

perrotta commented on RecoParticleFlow/PFClusterTools/src/PFEnergyCalibration.cc:
```diff
@@ -161,6 +161,59 @@ void PFEnergyCalibration::initializeCalibrationFunctions() {
+            //getStatusCode() == 0 => active channel
+            // apply correction if all recHits are dead
+            if (channelStatus.getMap().find(strip1)->getStatusCode() == 0)
+              condP1 = 0;
```
you can break here

perrotta commented on RecoParticleFlow/PFClusterTools/src/PFEnergyCalibration.cc:
```diff
@@ -161,6 +161,59 @@ void PFEnergyCalibration::initializeCalibrationFunctions() {
+          ESDetId strip2 = recH.recHitRef()->detId();
+          if (strip2 != ESDetId(0)) {
+            if (channelStatus.getMap().find(strip2)->getStatusCode() == 0)
+              condP2 = 0;
```
you can break here

perrotta commented on RecoEcal/EgammaClusterAlgos/src/PFECALSuperClusterAlgo.cc:
```diff
@@ -316,62 +316,25 @@ void PFECALSuperClusterAlgo::buildSuperCluster(CalibClusterPtr& seed, CalibClust
+      energyweighttot(0);
   for (auto& clus : clustered) {
-    ePS1 = ePS2 = 0;
+    double ePS1 = 0.0;
```
(just for symmetry) you could write "0" for both them...

guitargeek committed:
address review comments

cmsbuild commented:
The code-checks are being triggered in jenkins.

guitargeek commented:
Hi @perrotta, I addressed your comments and also got rid of an "temporaty ugly fix" in the PFEnergyCalibration class. Just reading the code of the PFCluster and CaloCluster classes, it seems equivalent to just use the `eta()` and `phi()` member functions without having to copy the cluster, but I guess the bot comparisons will tell if this is really correct.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-28960/13978

- This PR adds an extra 52KB to repository

cmsbuild commented:
Pull request #28960 was updated. @perrotta, @cmsbuild, @slava77 can you please check and sign again.

perrotta commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1

Tested at: 64c60c6a1ab9d4b8b941466aa7ed047d0d674378

CMSSW: CMSSW_11_1_X_2020-03-03-1100
SCRAM_ARCH: slc7_amd64_gcc820
You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ab1c14/4977/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals  AddOn**

* **RelVals**:

The relvals timed out after 4 hours.
When I ran the RelVals I found an error in the following workflows:
5.1 step1
<pre>runTheMatrix-results/5.1_TTbar+TTbarFS+HARVESTFS/step1_TTbar+TTbarFS+HARVESTFS.log

</pre>135.4 step1
<pre>runTheMatrix-results/135.4_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS/step1_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS.log

</pre>4.53 step3
<pre>runTheMatrix-results/4.53_RunPhoton2012B+RunPhoton2012B+HLTD+RECODR1reHLT+HARVESTDR1reHLT/step3_RunPhoton2012B+RunPhoton2012B+HLTD+RECODR1reHLT+HARVESTDR1reHLT.log

</pre>136.731 step3
<pre>runTheMatrix-results/136.731_RunSinglePh2016B+RunSinglePh2016B+HLTDR2_2016+RECODR2_2016reHLT_skimSinglePh_HIPM+HARVESTDR2/step3_RunSinglePh2016B+RunSinglePh2016B+HLTDR2_2016+RECODR2_2016reHLT_skimSinglePh_HIPM+HARVESTDR2.log

</pre>136.793 step3
<pre>runTheMatrix-results/136.793_RunDoubleEG2017C+RunDoubleEG2017C+HLTDR2_2017+RECODR2_2017reHLT_skimDoubleEG_Prompt+HARVEST2017/step3_RunDoubleEG2017C+RunDoubleEG2017C+HLTDR2_2017+RECODR2_2017reHLT_skimDoubleEG_Prompt+HARVEST2017.log

</pre>140.56 step2
<pre>runTheMatrix-results/140.56_RunHI2018+RunHI2018+RECOHID18+HARVESTDHI18/step2_RunHI2018+RunHI2018+RECOHID18+HARVESTDHI18.log

</pre>158.0 step2
<pre>runTheMatrix-results/158.0_HydjetQ_B12_5020GeV_2018_ppReco+HydjetQ_B12_5020GeV_2018_ppReco+DIGIHI2018PPRECO+RECOHI2018PPRECO+ALCARECOHI2018PPRECO+HARVESTHI2018PPRECO/step2_HydjetQ_B12_5020GeV_2018_ppReco+HydjetQ_B12_5020GeV_2018_ppReco+DIGIHI2018PPRECO+RECOHI2018PPRECO+ALCARECOHI2018PPRECO+HARVESTHI2018PPRECO.log

</pre>1000.0 step2
<pre>runTheMatrix-results/1000.0_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT/step2_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT.log

</pre>1001.0 step2
<pre>runTheMatrix-results/1001.0_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVDSIPIXELCALRUN1+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4+ALCAHARVD5/step2_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVDSIPIXELCALRUN1+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4+ALCAHARVD5.log

</pre>9.0 step3
<pre>runTheMatrix-results/9.0_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST/step3_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST.log

</pre>25.0 step3
<pre>runTheMatrix-results/25.0_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT/step3_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT.log

</pre>1306.0 step3
<pre>runTheMatrix-results/1306.0_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15/step3_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15.log

</pre>1330.0 step3
<pre>runTheMatrix-results/1330.0_ZMM_13+ZMM_13+DIGIUP15+RECOUP15_L1TMuDQM+HARVESTUP15_L1TMuDQM+NANOUP15/step3_ZMM_13+ZMM_13+DIGIUP15+RECOUP15_L1TMuDQM+HARVESTUP15_L1TMuDQM+NANOUP15.log

</pre>10042.0 step3
<pre>runTheMatrix-results/10042.0_ZMM_13+ZMM_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017+ALCAFull_2017+NanoFull_2017/step3_ZMM_13+ZMM_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017+ALCAFull_2017+NanoFull_2017.log

</pre>11634.0 step2
<pre>runTheMatrix-results/11634.0_TTbar_14TeV+TTbar_14TeV_TuneCP5_2021_GenSimFull+DigiFull_2021+RecoFull_2021+HARVESTFull_2021+ALCAFull_2021/step2_TTbar_14TeV+TTbar_14TeV_TuneCP5_2023_GenSimFull+DigiFull_2023+RecoFull_2023+HARVESTFull_2023+ALCAFull_2023.log

</pre>12434.0 step2
<pre>runTheMatrix-results/12434.0_TTbar_14TeV+TTbar_14TeV_TuneCP5_2023_GenSimFull+DigiFull_2023+RecoFull_2023+HARVESTFull_2023+ALCAFull_2023/step2_TTbar_14TeV+TTbar_14TeV_TuneCP5_2023_GenSimFull+DigiFull_2023+RecoFull_2023+HARVESTFull_2023+ALCAFull_2023.log

</pre>10024.0 step3
<pre>runTheMatrix-results/10024.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017+ALCAFull_2017+NanoFull_2017/step3_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017+ALCAFull_2017+NanoFull_2017.log

</pre>25202.0 step3
<pre>runTheMatrix-results/25202.0_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25+NANOUP15_PU25/step3_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25+NANOUP15_PU25.log

</pre>10824.0 step3
<pre>runTheMatrix-results/10824.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2018_GenSimFull+DigiFull_2018+RecoFull_2018+HARVESTFull_2018+ALCAFull_2018+NanoFull_2018/step3_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2018_GenSimFull+DigiFull_2018+RecoFull_2018+HARVESTFull_2018+ALCAFull_2018+NanoFull_2018.log

</pre>10224.0 step3
<pre>runTheMatrix-results/10224.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017PU_GenSimFull+DigiFullPU_2017PU+RecoFullPU_2017PU+HARVESTFullPU_2017PU+NanoFull_2017PU/step3_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017PU_GenSimFull+DigiFullPU_2017PU+RecoFullPU_2017PU+HARVESTFullPU_2017PU+NanoFull_2017PU.log

</pre>20034.0 step3
<pre>runTheMatrix-results/20034.0_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D35_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D35+RecoFullGlobal_2026D35+HARVESTFullGlobal_2026D35/step3_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D35_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D35+RecoFullGlobal_2026D35+HARVESTFullGlobal_2026D35.log

</pre>20434.0 step3
<pre>runTheMatrix-results/20434.0_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D41_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D41+RecoFullGlobal_2026D41+HARVESTFullGlobal_2026D41/step3_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D41_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D41+RecoFullGlobal_2026D41+HARVESTFullGlobal_2026D41.log

</pre>21234.0 step3
<pre>runTheMatrix-results/21234.0_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D44_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D44+RecoFullGlobal_2026D44+HARVESTFullGlobal_2026D44/step3_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D49_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D49+RecoFullGlobal_2026D49+HARVESTFullGlobal_2026D49.log

</pre>

* **AddOn**:

I found errors in the following addon tests:

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-28960/13987

- This PR adds an extra 52KB to repository

cmsbuild commented:
Pull request #28960 was updated. @perrotta, @cmsbuild, @slava77 can you please check and sign again.

guitargeek commented:
Hi @perrotta, I made a mistake in my previous commit. Now I fixed that commit and also made sure the local tests pass. So the bot should succeed now!

perrotta commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: a932054154b76b79627bbec6714a36743a04bc14
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ab1c14/4988/summary.html
CMSSW: CMSSW_11_1_X_2020-03-03-2300
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ab1c14/4988/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 34
 - DQMHistoTests: Total histograms compared: 2680373
 - DQMHistoTests: Total failures: 39
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2680015
 - DQMHistoTests: Total skipped: 319
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 33 files compared)
 - Checked 147 log files, 16 edm output root files, 34 DQM output files

cmsbuild commented:
The code-checks are being triggered in jenkins.

guitargeek commented:
Hi @perrotta, inspired by your previous comments that suggested to use `bool` and not `double` to encode whether the PS planes should be used for the corrections or not, I propagated this change now through all the functions of the class. That makes it more understandable, as using the same variables to declare the PS plane status and store the output was confusing.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-28960/14014

- This PR adds an extra 56KB to repository

cmsbuild commented:
Pull request #28960 was updated. @perrotta, @cmsbuild, @slava77 can you please check and sign again.

perrotta commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 6f6694722cac486fd7e455bd570a5561adeeb3d5
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ab1c14/5006/summary.html
CMSSW: CMSSW_11_1_X_2020-03-05-1100
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ab1c14/5006/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 1530 differences found in the comparisons
 - DQMHistoTests: Total files compared: 34
 - DQMHistoTests: Total histograms compared: 2680577
 - DQMHistoTests: Total failures: 4859
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2675399
 - DQMHistoTests: Total skipped: 319
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 33 files compared)
 - Checked 147 log files, 16 edm output root files, 34 DQM output files

perrotta commented:
> * Reco comparison results: 1530 differences found in the comparisons

The latest commit wasn't that innocuos... We must check where those differences come from.
Perhaps, factorizing the updates whenever possible could be a better option: it would at least be easier to review

guitargeek commented:
Yes that's right, it would have been nice if it worked but I tried to do too much at once at this PR. I will amend the last commit and only keep the changes that are hopefully really innocent, and then maybe we can do the rest at some other time.

guitargeek committed:
make one of the energyEm functions private

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-28960/14022

- This PR adds an extra 52KB to repository

cmsbuild commented:
Pull request #28960 was updated. @perrotta, @cmsbuild, @slava77 can you please check and sign again.

guitargeek commented:
Hi @perrotta, now it's ready to be tested again. I only kept the change with the other overloaded version of the `energyEm` function in the `ConvBremPFTrackFinder` function, so that the interface of `PFEnergyCalibration` needs fewer public functions. This should not change the result.

perrotta commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 6b13a9517b8bfe175d45bc642af7a175ee6208ff
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ab1c14/5039/summary.html
CMSSW: CMSSW_11_1_X_2020-03-05-2300
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ab1c14/5039/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 2 differences found in the comparisons
 - DQMHistoTests: Total files compared: 34
 - DQMHistoTests: Total histograms compared: 2680577
 - DQMHistoTests: Total failures: 40
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2680218
 - DQMHistoTests: Total skipped: 319
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 33 files compared)
 - Checked 147 log files, 16 edm output root files, 34 DQM output files

perrotta commented:
+1
- It allows some refactorization of the PF energy calibration 
- Jenkins tests pass and show no difference with the current reco outputs, as it should

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @davidlange6, @silviodonato, @fabiocos (and backports should be raised in the release meeting by the corresponding L2)

silviodonato commented:
+1

slava77 commented:
[I'm not sure if this is the PR with the problem, but the error messages appear to point to this]

@smuzaffar @mrodozov 
please remind me how to test on PPC interactively.
I suspect that the problem came from this PR, based on 140.52 working in CMSSW_11_1_X_2020-03-08-2300 but not in CMSSW_11_1_X_2020-03-09-2300.
In the latter we have
```
An exception of category 'StdException' occurred while
   [0] Processing  stream begin Run run: 152698 stream: 0
   [1] Calling method for module PFProducer/'particleFlowTmp'
   [2] Using EventSetup component PoolDBESSource/'GlobalTag' to make data PerformancePayload/'' in record PFCalibrationRcd
Exception Message:
A std::exception was thrown.
getrandom
```

https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/slc7_ppc64le_gcc820/CMSSW_11_1_X_2020-03-08-2300/pyRelValMatrixLogs/run/140.52_RunHI2010+RunHI2010+RECOHID10+RECOHIR10D11+HARVESTDHI/step2_RunHI2010+RunHI2010+RECOHID10+RECOHIR10D11+HARVESTDHI.log
vs
https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/slc7_ppc64le_gcc820/CMSSW_11_1_X_2020-03-09-2300/pyRelValMatrixLogs/run/140.52_RunHI2010+RunHI2010+RECOHID10+RECOHIR10D11+HARVESTDHI/step2_RunHI2010+RunHI2010+RECOHID10+RECOHIR10D11+HARVESTDHI.log

slava77 commented:
ping

https://github.com/cms-sw/cmssw/pull/28960#issuecomment-597599381 
> @smuzaffar @mrodozov
please remind me how to test on PPC interactively.

smuzaffar commented:
@slava77 send me your ssh public key

slava77 commented:
I managed to get to the error in PerformancePayload.
It is apparently unrelated to this PR.
The same IB where the problem appeared apparently also has https://github.com/cms-sw/cmsdist/pull/5573
the gdb stack trace points to a problem in boost and simply merging this PR on top of 08-2300 IB does not introduce any problems.
